cmd/edit/machinepool/nodepool.go

```diff
@@ -307,12 +307,25 @@ func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int)
 	asBuilder := cmv1.NewNodePoolAutoscaling()
 	changed := false
 
-	if nodePool.Autoscaling().MinReplica() != minReplicas && minReplicas >= 1 {
-		asBuilder = asBuilder.MinReplica(minReplicas)
+	existingMinReplica := nodePool.Autoscaling().MinReplica()
+	existingMaxReplica := nodePool.Autoscaling().MaxReplica()
+
+	if existingMinReplica != minReplicas {
+		if minReplicas >= 1 {
+			asBuilder = asBuilder.MinReplica(minReplicas)
+		}
+		if minReplicas == 0 {
+			asBuilder = asBuilder.MinReplica(existingMinReplica)
+		}
 		changed = true
 	}
-	if nodePool.Autoscaling().MaxReplica() != maxReplicas && maxReplicas >= 1 {
-		asBuilder = asBuilder.MaxReplica(maxReplicas)
+	if existingMaxReplica != maxReplicas {
+		if maxReplicas >= 1 {
```

Comment:
same as above, `changed = true` should not be here if maxReplicas not set?

Reply:
Thanks @marcolan018 , updated with the change